avm/ptn/lz/sub-vending/src/self/subResourceWrapper/deploy.bicep

```diff
@@ -0,0 +1,689 @@
+
+> ⚠️ It is not intended for this module to be called outside of being a sub-orchestration module for the `main.bicep` module ⚠️'''
+
+// PARAMETERS
+
+@maxLength(36)
+param subscriptionId string
+
+@sys.description('Whether to move the subscription to the specified management group supplied in the pararmeter subscriptionManagementGroupId.')
+param subscriptionManagementGroupAssociationEnabled bool = true
+
+@sys.description('The destination management group ID for the new subscription. Note: Do not supply the display name. The management group ID forms part of the Azure resource ID. e.g., `/providers/Microsoft.Management/managementGroups/{managementGroupId}`.')
+param subscriptionManagementGroupId string = ''
+
+@sys.description('An object of tag key/value pairs to be appended to a subscription. NOTE: Tags will only be overwriten if existing tag exists with same key; values provided here win.')
+param subscriptionTags object = {}
+
+@sys.description('Whether to create a virtual network or not.')
+param virtualNetworkEnabled bool = false
+
+@maxLength(90)
+@sys.description('The name of the resource group to create the virtual network in.')
+param virtualNetworkResourceGroupName string = ''
+
+@sys.description('Enables the deployment of a `CanNotDelete` resource locks to the virtual networks resource group.')
+param virtualNetworkResourceGroupLockEnabled bool = true
+
+@sys.description('An object of tag key/value pairs to be appended to the Resource Group that the Virtual Network is created in. NOTE: Tags will only be overwriten if existing tag exists with same key; values provided here win.')
+param virtualNetworkResourceGroupTags object = {}
+
+@sys.description('The location of the virtual network. Use region shortnames e.g. uksouth, eastus, etc.')
+param virtualNetworkLocation string = deployment().location
+
+@maxLength(64)
+@sys.description('The name of the virtual network. The string must consist of a-z, A-Z, 0-9, -, _, and . (period) and be between 2 and 64 characters in length.')
+param virtualNetworkName string = ''
+
+@sys.description('An object of tag key/value pairs to be set on the Virtual Network that is created. NOTE: Tags will be overwritten on resoruce if any exist already.')
+param virtualNetworkTags object = {}
+
+@sys.description('The address space of the virtual network, supplied as multiple CIDR blocks, e.g. `["10.0.0.0/16","172.16.0.0/12"]`')
+param virtualNetworkAddressSpace array = []
+
+@sys.description('The custom DNS servers to use on the virtual network, e.g. `["10.4.1.4", "10.2.1.5"]. If left empty (default) then Azure DNS will be used for the virtual network.`')
+param virtualNetworkDnsServers array = []
+
+@sys.description('The resoruce ID of an existing DDoS Network Protection Plan that you wish to link to this virtual network.')
+param virtualNetworkDdosPlanId string = ''
```

Comment:
Should be `virtualNetworkDdosPlanResourceId`

Reply:
Resolved